src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -741,6 +768,12 @@ - (void)_handleReportEnd
     _receivingReport = NO;
     _receivingPrimingReport = NO;
     _estimatedStartTimeFromGeneralDiagnosticsUpTime = nil;
+
+    // First subscription report is priming report
```

Comment:
Shouldn't this go in the "subscription established" callback, instead of guessing things based on "report end"?

Reply:
Good point. I've moved the logic there.